src/Resources/ResourceManager/Utilities/BicepUtility.cs

```diff
@@ -153,7 +153,7 @@ public void PublishFile(string bicepFilePath, string target, string documentatio
             if (!string.IsNullOrWhiteSpace(documentationUri))
             {
                 CheckMinimalVersionRequirement(MinimalVersionRequirementForBicepPublishWithOptionalDocumentationUriParameter);
-                bicepPublishCommand += $" --documentationUri {GetQuotedFilePath(documentationUri)}";
+                bicepPublishCommand += $" --documentation-uri {GetQuotedFilePath(documentationUri)}";
```

Comment:
To avoid breaking users of older versions of Bicep, we'll need to only use the new format if the version of Bicep is greater than or equal to [0.24.24](https://github.com/Azure/bicep/blob/v0.24.24/src/Bicep.Cli/Arguments/PublishArguments.cs), and retain the old logic if not.

You could probably use a check similar to https://github.com/Azure/azure-powershell/blob/d20265b6a4c6d82c3048c8728b2fb97992772cfe/src/Resources/ResourceManager/Utilities/BicepUtility.cs#L191 to accomplish this.